src/awkward/_nplikes/array_module.py

```diff
@@ -53,15 +53,11 @@ def ascontiguousarray(self, x: ArrayLike) -> ArrayLike:
             return self._module.ascontiguousarray(x)
 
     def frombuffer(
-        self, buffer, *, dtype: np.dtype | None = None, count: int = -1
+        self, buffer, *, dtype: np.dtype | None = None, count: ShapeItem = -1
     ) -> ArrayLike:
         if isinstance(buffer, PlaceholderArray):
-            if count == -1:
-                return self.asarray(buffer)
-            else:
-                return self.asarray(buffer[:count])
-        else:
-            return self._module.frombuffer(buffer, dtype=dtype, count=count)
+            raise TypeError("placeholder arrays are not supported in `frombuffer`")
+        return self._module.frombuffer(buffer, dtype=dtype, count=count)
```

Comment:
I don't think we should support placeholders in `nplike.frombuffer`; they don't actually have any data. We could make exceptions to this rule, but for `frombuffer` this doesn't make any sense.

Reply:
I agree: even though we'd be loading the placeholder at the point where we normally call `frombuffer`, having that pass through `frombuffer` is an abuse of the name.